### PR TITLE
Preserve original error when there is no fallback formatter

### PR DESCRIPTION
**Pull request by CrabDude**
## Pre-Submission Checklist

- [x] Ran the linter and tests via `make prepush`
- [x] Included comprehensive and convincing tests for changes

## Issues

> Summarize the issues that discussed these changes
There is an edge case where `strictFormatters=true`, `content-type` is set without a supported formatter, and an error is passed. Right now the error will fall through to a raw serialization (`res.write`), which fails to serialize the error object.

# Changes

> What does this PR do?

Ensures the original error is preserved with an `InternalServerError`, instead of falling through to `res.write`.

### CODE REVIEW

File: lib/response.js [hunk 1]
```diff
@@ -529,6 +529,19 @@ function patch(Response) {
                 }
                 return flush(self, formattedBody);
             }
+            // Preserve original error when
+            // Content-type is set, _strictFormatters=false, no formatter found
+            if (body instanceof Error) {
```
Comment (DonutEspresso) [inline lib/response.js]: Couldn’t this be something other than an error? Maybe check against what we assert in `flush()`?

Comment (retrohacker) [reply lib/response.js]: I don't understand @DonutEspresso, isn't this PR specifically for preserving errors?

Comment (DonutEspresso) [reply lib/response.js]: It's been awhile since I looked at this - I _think_ my question was if it was possible to get to this point in the function where body isn't an Error but just a POJO. IOW, does this new `if` clause cause a silent failure/hang where there's no formatter, and the body is not an Error?

Comment (DonutEspresso) [reply lib/response.js]: Maybe there's already a test for this in which case carry on!
